hkmc2/shared/src/main/scala/hkmc2/bbml/ConstraintSolver.scala

```diff
@@ -117,12 +120,21 @@ class ConstraintSolver(infVarState: InfVarUid.State, tl: TraceLogger):
   private def constrainArgs(lhs: TypeArg, rhs: TypeArg)(using Ctx, CCtx, TL): Unit =
     constrainImpl(rhs.negPart, lhs.negPart)
     constrainImpl(lhs.posPart, rhs.posPart)
-  
+
+  private def inlineSkolemBounds(ty: Type, pol: Bool)(using cache: Set[Uid[InfVar]]): Type = ty match
+    case v @ InfVar(_, uid, state, skolem) if skolem && !cache(uid) =>
+      given Set[Uid[InfVar]] = cache + uid
+      inlineSkolemBounds(if pol then (v :: state.upperBounds).reduceLeft(_ & _) else (v :: state.lowerBounds).reduceLeft(_ | _), pol)
```

Comment:
The "reduce" functions should be avoided. It makes a lot more sense to use `foldLeft` here.